docs/design/services/s3/transfermanager/prototype.java

```diff
@@ -0,0 +1,721 @@
+package software.amazon.awssdk.services.s3.transfer;
+
+/**
+ * The S3 Transfer Manager is a library that allows users to easily and
+ * optimally upload and downloads to and from S3.
+ * <p>
+ * The list of features includes:
+ * <ul>
+ *   <li>Parallel uploads and downloads</li>
+ *   <li>Bandwidth limiting</li>
+ *   <li>Pause and resume of transfers</li>
+ * </ul>
+ * <p>
+ * <b>Usage Example:</b>
+ * <pre>
+ * {@code
+ * // Create using all default configuration values
+ * S3TransferManager tm = S3TranferManager.create();
+ *
+ * // Using custom configuration values to set max upload speed to avoid
+ * // saturating the network interface.
+ * S3TransferManager tm = S3TransferManager.builder()
+ *         .configuration(TransferManagerConfiguration.builder()
+ *             .maximumUploadBytesSecond(32 * 1024 * 1024) // 32 MiB
```

Comment:
Can we expose a minimum as well?

Additionally, can we remove "Upload" and make it generic for both?

Reply:
Hmm how would minimum work at the TransferManager level? Seems like it might more sense at the individual transfer level.

I'm not sure about a single config for both upload and download. Seems to me like there might be cases where you don't want them to be symmetric; maybe the server is serving other traffic so can't dedicate all outbound bandwidth to uploads for example.